src/dmd/backend/debugprint.d

```diff
@@ -0,0 +1,486 @@
+
+                if (b && b.Bweight)
+                        printf("B%d (%p) ",b.Bdfoidx,b);
+                else
+                        printf("%p ",b);
+        }
+        ferr("\n");
+}
+
+void WRdefnod()
+{ int i;
+
+  for (i = 0; i < go.deftop; i++)
+  {     printf("defnod[%d] in B%d = (", go.defnod[i].DNblock.Bdfoidx, i);
+        WReqn(go.defnod[i].DNelem);
+        printf(");\n");
+  }
+}
+
+void WRFL(FL fl)
+{
+    __gshared const(char)[7][FLMAX] fls =
+    [    "unde  ","const ","oper  ","func  ","data  ",
+         "reg   ",
+         "pseudo",
+         "auto  ","fast  ","para  ","extrn ",
+         "code  ","block ","udata ","cs    ","swit  ",
+         "fltrg ","offst ","datsg ",
+         "ctor  ","dtor  ","regsav","asm   ",
+         "ndp   ",
+         "farda ","csdat ",
+         "local ","tlsdat",
+         "bprel ","frameh","blocko","alloca",
+         "stack ","dsym  ",
+         "got   ","gotoff",
+         "funcar",
+    ];
+
+    if (cast(uint)fl >= FLMAX)
+        printf("FL%d",fl);
+    else
+      printf("FL%s",fls[fl].ptr);
+}
+
+/***********************
+ * Write out block.
+ */
+
+void WRblock(block *b)
+{
+    if (OPTIMIZER)
+    {
+        if (b && b.Bweight)
+                printf("B%d: (%p), weight=%d",b.Bdfoidx,b,b.Bweight);
+        else
+                printf("block %p",b);
+        if (!b)
+        {       ferr("\n");
+                return;
+        }
+        printf(" flags=x%x weight=%d",b.Bflags,b.Bweight);
+        //printf("\tfile %p, line %d",b.Bfilptr,b.Blinnum);
+        printf(" ");
+        WRBC(b.BC);
+        printf(" Btry=%p Bindex=%d",b.Btry,b.Bindex);
+        if (b.BC == BCtry)
+            printf(" catchvar = %p",b.catchvar);
+        printf("\n");
+        printf("\tBpred: "); WRblocklist(b.Bpred);
+        printf("\tBsucc: "); WRblocklist(b.Bsucc);
+        if (b.Belem)
+        {       if (debugf)                     /* if full output       */
+                        elem_print(b.Belem);
+                else
+                {       ferr("\t");
+                        WReqn(b.Belem);
+                        printf(";\n");
+                }
+        }
+        version (MARS)
+        {
+        if (b.Bcode)
+            b.Bcode.print();
+        }
+        version (SCPP)
+        {
+        if (b.Bcode)
+            b.Bcode.print();
+        }
+        ferr("\n");
+    }
+    else
+    {
+        targ_llong *pu;
+        int ncases;
+
+        assert(b);
+        printf("%2d: ", b.Bnumber); WRBC(b.BC);
+        if (b.Btry)
+            printf(" Btry=B%d",b.Btry ? b.Btry.Bnumber : 0);
+        if (b.Bindex)
+            printf(" Bindex=%d",b.Bindex);
+        if (b.BC == BC_finally)
+            printf(" b_ret=B%d", b.b_ret ? b.b_ret.Bnumber : 0);
+version (MARS)
+{
+        if (b.Bsrcpos.Sfilename)
+            printf(" %s(%u)", b.Bsrcpos.Sfilename, b.Bsrcpos.Slinnum);
+}
+        printf("\n");
+        if (b.Belem)
+        {
+            if (debugf)
```

Comment:
First rule of conversion ;-) (it's originally `0`)

Reply:
Eh, must be my copy of debug.c differed slightly. No matter.